src/js/components/Pagination/Pagination.js

```diff
@@ -263,7 +263,7 @@ const Pagination = forwardRef(
         <Box
           align="center"
           direction="row"
-          gap={{ column: 'xsmall', row: 'small' }}
+          gap={theme.pagination.withControls?.gap}
```

Comment:
would there be a case in which someone would want a different `gap` here than the container gap?

Reply:
ignore this comment, look again at if we can use the theme.pagination.container.gap if that causes snapshot changes

Reply:
Just checked and using `theme.pagination.container.gap` is causing snapshot changes
<img width="569" height="225" alt="Screenshot 2025-08-11 at 11 11 10 AM" src="https://github.com/user-attachments/assets/39803e52-d02f-46e9-8f94-93c43978d785" />

Reply:
Can we do this on ~ line 230:

```
// for backwards compatibility
    if (!summary && !stepOptions)
      return (
        <StyledPaginationContainer
          flex={false}
          // Set gap to undefined for backwards compatibility
          // There is only one child anyway, so gap would not have any effect
          {...{ ...theme.pagination.container, gap: undefined }}
          {...passThemeFlag}
          {...rest}
        >
          {paginationControls}
        </StyledPaginationContainer>
      );
```

It avoids any snapshot changes (even though the snapshot changes due to gap wouldn't have effected the layout anyway given that container only has a single child.

Then we don't have to introduce a `withControls` object and can just use the existing.